src/plugin/context.rs

```diff
@@ -209,19 +208,23 @@ impl RapierContext {
         &mut self,
         gravity: Vect,
         timestep_mode: TimestepMode,
-        events: Option<(EventWriter<CollisionEvent>, EventWriter<ContactForceEvent>)>,
+        create_bevy_events: bool,
         hooks: &dyn PhysicsHooks,
         time: &Time,
         sim_to_render_time: &mut SimulationToRenderTime,
-        mut interpolation_query: Option<
-            Query<(&RapierRigidBodyHandle, &mut TransformInterpolation)>,
+        interpolation_query: &mut Option<
+            &mut Query<(&RapierRigidBodyHandle, &mut TransformInterpolation)>,
         >,
     ) {
-        let event_queue = events.map(|(ce, fe)| EventQueue {
-            deleted_colliders: &self.deleted_colliders,
-            collision_events: RwLock::new(ce),
-            contact_force_events: RwLock::new(fe),
-        });
+        let event_queue = if create_bevy_events {
+            Some(EventQueue {
+                deleted_colliders: &self.deleted_colliders,
+                collision_events: &mut self.collision_events_to_send,
+                contact_force_events: &mut self.contact_force_events_to_send,
```

Comment:
I also agree that adding a PhysicsWorld field to those events would be a good idea, even though you could just query it in the event handlers.  Since it's just a usize there would be basically no overhead and make it more convenient for some use cases.